### DISCUSSION TITLE
Gui for 7in Raspberry Pi touchscreen

### DISCUSSION THREAD

**Discussion by thecharacterg** [GeekFunkLabs/fluidpatcher#9] [Ideas]
Starting to work on this adaption of script to have the touchscreen start with doing the same thing that the squishbox does. I seen the calls to the lcd screen in one part, but to admit having an issue of finding where to edit once it's installed in the pi. What file am I going to be editing?

**Comment by albedozero** [comment]
A better place to start might be to look at the simple example implementation and explanation in the [patcher API](https://github.com/albedozero/fluidpatcher/blob/master/patcher/README.md). You _could_ start by just changing some of the `sb.lcd_write` calls in _squishbox.py_ to `print`s, but the way you display information and create information on a touch screen is so different from a little 16x2 character display that I don't think it would be that instructive or worth your time. The _fluidpatcher.pyw_ GUI program is screen-based and might help a bit more, but I'd suggest looking at the API and some touchscreen coding examples if you haven't already.

**Reply by thecharacterg** [reply]
To be honest, I seen it and you’re over thinking what I am trying to do. It’s creating a GUI period that does what the stomp box does. All touch operations are done by the pi.

**Reply by albedozero** [reply]
I see - but there is a lot of overhead to creating a touchscreen GUI app that is different from that for an app that interacts with physical buttons and a tiny text screen. Instead of just checking every few milliseconds to see whether the voltage from a button is high or low, for example, your app has to draw a box on the screen for your button, then set up a callback function in an event loop that is triggered when someone touches the screen, which then checks to see if the location of the touch is within the box. You'd probably want to create separate buttons for the menus (for loading banks, saving patches, etc.) rather than make the user hold their finger on the patch-change buttons for >2 seconds to open them. You probably will end up using _squishbox.py_ as the bones for your app, to handle the files and calls to Fluidsynth, etc., but the interface part will be very different.

Have you done GUI programming for the touchscreen? I've written GUI apps for mouse+keyboard using Python, but I don't have a touchscreen to mess around with. [Kivy](https://kivy.org/doc/stable/gettingstarted/intro.html) seems like a Python library/framework that would work pretty well for a touchscreen app like this, but I don't have a way to try it out at the moment.
